# SF-DAT-21 | Unit Project 4

In this project, you will summarize and present your analysis from Unit Projects 1-3.

### Question 1. Introduction: Write a problem Statement/Specific Aim for this project.

Answer: Using the Admissions dataset from a UCLA Graduate Program, we went to understand what effect undergraduate GPE, GRE Score and Alma Mater prestige have on admissions. Ideally we would like to understand what factors are most important in predicting this. Due to the Predicted Variable being categorical, we should use a classification model

In [9]:
import pandas as pd

df = pd.read_csv('../../Dataset/admissions.csv')
df = df.dropna()


dummy_ranks = pd.get_dummies(df.prestige, prefix = 'prestige').astype(int)
df = df.join([dummy_ranks])

df.describe()
df.groupby('admit').describe()

gpa         gre    prestige  prestige_1.0  prestige_2.0  \
admit                                                                         
0     count  271.000000  271.000000  271.000000    271.000000    271.000000   
      mean     3.347159  573.579336    2.645756      0.103321      0.350554   
      std      0.376355  116.052798    0.918922      0.304941      0.478026   
      min      2.260000  220.000000    1.000000      0.000000      0.000000   
      25%      3.080000  500.000000    2.000000      0.000000      0.000000   
      50%      3.340000  580.000000    3.000000      0.000000      0.000000   
      75%      3.610000  660.000000    3.000000      0.000000      1.000000   
      max      4.000000  800.000000    4.000000      1.000000      1.000000   
1     count  126.000000  126.000000  126.000000    126.000000    126.000000   
      mean     3.489206  618.571429    2.150794      0.261905      0.420635   
      std      0.371655  109.257233    0.921455      0.441426      0.495632   
      min      2.420000  300.000000    1.000000      0.000000      0.000000   
      25%      3.220000  540.000000    1.000000      0.000000      0.000000   
      50%      3.545000  620.000000    2.000000      0.000000      0.000000   
      75%      3.757500  680.000000    3.000000      1.000000      1.000000   
      max      4.000000  800.000000    4.000000      1.000000      1.000000   

             prestige_3.0  prestige_4.0  
admit                                    
0     count    271.000000    271.000000  
      mean       0.343173      0.202952  
      std        0.475647      0.402941  
      min        0.000000      0.000000  
      25%        0.000000      0.000000  
      50%        0.000000      0.000000  
      75%        1.000000      0.000000  
      max        1.000000      1.000000  
1     count    126.000000    126.000000  
      mean       0.222222      0.095238  
      std        0.417399      0.294715  
      min        0.000000      0.000000  
      25%        0.000000      0.000000  
      50%        0.000000      0.000000  
      75%        0.000000      0.000000  
      max        1.000000      1.000000

In [10]:
pd.crosstab(df.prestige, df.admit, margins=True)

admit       0    1  All
prestige               
1.0        28   33   61
2.0        95   53  148
3.0        93   28  121
4.0        55   12   67
All       271  126  397

In [11]:
df.corr()

admit       gre       gpa  prestige  prestige_1.0  \
admit         1.000000  0.181202  0.174116 -0.243563      0.204689   
gre           0.181202  1.000000  0.382408 -0.124533      0.088277   
gpa           0.174116  0.382408  1.000000 -0.060976      0.068304   
prestige     -0.243563 -0.124533 -0.060976  1.000000     -0.670582   
prestige_1.0  0.204689  0.088277  0.068304 -0.670582      1.000000   
prestige_2.0  0.067459  0.058454 -0.050507 -0.398292     -0.328493   
prestige_3.0 -0.122302 -0.074380  0.070881  0.357934     -0.282120   
prestige_4.0 -0.133859 -0.069046 -0.087671  0.719947     -0.191989   

              prestige_2.0  prestige_3.0  prestige_4.0  
admit             0.067459     -0.122302     -0.133859  
gre               0.058454     -0.074380     -0.069046  
gpa              -0.050507      0.070881     -0.087671  
prestige         -0.398292      0.357934      0.719947  
prestige_1.0     -0.328493     -0.282120     -0.191989  
prestige_2.0      1.000000     -0.510469     -0.347385  
prestige_3.0     -0.510469      1.000000     -0.298345  
prestige_4.0     -0.347385     -0.298345      1.000000

### Question 2. Dataset: Write up a description of your data and any cleaning that was completed.


Answer: 
1. 400 Observations with 4 variables:
    -admit (1, 0)
    -GRE (continuous)
    -GPA (continuous)
    -Prestige (1, 2, 3, 4)
    
2. We create dummy variables for Prestige to help understand how Prestige predicts Admission
3. 3 records with missing values (which we drop)
4. Admission Rate ~ 30% on the whole
5. Higher GPA and GRE Scores for admitted students
6. Prestige 1 schools appear to have a higher admittance rate than 2, 3, 4

### Question 3. Demo: Provide a table that explains the data by admission status.

Mean (STD) or counts by admission status for each variable

| Not Admitted | Admitted
---| ---|---
GPA | 3.35(.376) | 3.49(.372)
GRE |574(116) | 619(109)
Prestige 1 | 28 | 33 
Prestige 2 | 95 | 53 
Prestige 3 |93  | 28 
Prestige 4 |55  | 12 

### Question 4. Methods: Write up the methods used in your analysis.

Answer: Since the predicted variable is binary/ categorical, we choose to run a logistic regression

### Question 5. Results: Write up your results.

Answer: The model score is 71%, which means we classify 71% of observations correctly. By calculating several odds ratios we are able to assertain some findings about various predictors-

1. Prestige & GPA matter!
    - The odds of being admitted and not coming from a rank 1-3 school is 2.4 times the odds of being admitted from a rank 4
    - The odds of admittance from a rank 2 school is 50% that of a rank 1 school
    - The odds of getting admitted with a higher GPA is >2X the odds of getting in with a lower GPA (lower by one increment?)

### Question 6. Visuals: Provide a table or visualization of these results.

<img src='../../../../OR.png'>

<img src='../../../../logit.png'>

### question 7. Discussion: Write up your discussion and future steps.

Answer: As a next step we would like to re-run the analysis using cross validation. Splitting out a test and train dataset